### DISCUSSION TITLE
script name conflicts with usbutils lsusb

### DISCUSSION THREAD

**Discussion by tormodvolden** [LanikSJ/lsusb#34] [Ideas]
### Description

The homebrew package _usbutils_ ships the "real" lsusb binary that is the same as on Linux. Since it would be great if people could have both this and your script installed and run them without confusion about what is running, would you please consider renaming your script to something else, for instance lsusb-plus? Or "llusb", just suggestions.

### Reproduction steps

```bash
1. Try to install both usbutils and lsusb-laniksj
2. Package manager will refuse to have both installed
3. Install one of them
4. Forget
5. Run "lsusb" and you don't know exactly what is running.
6. Somebody suggest you to run lsusb to provide debug information
7. You run "lsusb" but it happens to be the script, with incompatible output
```

### Screenshots

```bash
![DESCRIPTION](LINK.png)
```

### OS

Mac

**Comment by LanikSJ** [comment]
My tap already contains code that prevents both binaries from being installed with Homebrew.

```
  conflicts_with "lsusb", because: "both provide an `lsusb` binary"
  conflicts_with "lsusb-laniksj", because: "both provide an `lsusb` binary"
  conflicts_with "usbutils", because: "both provide an `lsusb` binary"
```
If you attempt to install `lsusb` with `usbutils` already installed you'll get an error and vise versa.

I'm not sure of a use case when you'd need both of them.

**Comment by tormodvolden** [comment]
Yes, both packages have conflicts declared so they cannot both be installed. That was point 2 in the Reproduction steps.

Well, they are two different programs, with different output when you run them. The one from usbutils is certainly useful, since it provides output compatible with the real lsusb on Linux, it is a cross-platform tool working the same on several platforms. I am not sure about the use case for your script, but I guess it may display some information that is not seen in lsusb. What confuses people is that it is called lsusb. It is great that the package is not called "lsusb" any longer, because that was even more confusing. If you also could rename the command, people could freely use both without confusion.

**Comment by tormodvolden** [comment]
I also don't know why you talk about your tap. It is an official homebrew package now: https://formulae.brew.sh/formula/lsusb-laniksj

**Comment by LanikSJ** [comment]
> Well, they are two different programs, with different output when you run them. The one from usbutils is certainly useful, since it provides output compatible with the real lsusb on Linux, it is a cross-platform tool working the same on several platforms. 

Yes no doubt it's usefulness on Linux.  Less so on a Mac since it doesn't work when I just installed and tested it.

>I am not sure about the use case for your script, but I guess it may display some information that is not seen in lsusb. What confuses people is that it is called lsusb. It is great that the package is not called "lsusb" any longer, because that was even more confusing. If you also could rename the command, people could freely use both without confusion.

If you want to see the listing of all your USB devices that's what the tool was designed to do.

For time being I'd leave things the way they are as you can't really run both tools on a Mac.  Now if that ever changes I could be persuaded to change my mind but right now I don't see a use case where this would ever be a problem.  Folks would either have one tool installed or the other.

>I also don't know why you talk about your tap. It is an official homebrew package now: https://formulae.brew.sh/formula/lsusb-laniksj

I'm well aware of that but I'm not a maintainer of Homebrew so my voice has very little sway in what they decide.  The community decided it was helpful to include this fork in Homebrew and obviously I have no objections. 🙂

The only reason I bring it up is because I can make changes to my tap a hell of a lot faster then Homebrew maintainers can.

**Comment by LanikSJ** [comment]
I moved this to a discussion as this doesn't really qualify as a bug any longer.

**Comment by tormodvolden** [comment]
Yes, if usbutils doesn't work, that's a more important problem!

What was the architecture and version that you tested on? What was the error or output?

**Reply by LanikSJ** [reply]
> What was the architecture and version that you tested on? What was the error or output?

Mac arm64 M2.

No error no output either just a blank line:

```bash
→ lsusb

```

**Comment by trinitronx** [comment]
> I'm well aware of that but I'm not a maintainer of Homebrew so my voice has very little sway in what they decide.

Hopefully a helpful bit of info:  Homebrew is community maintained, and their CI/CD system is highly automated.  So much so that nowadays it's very easy for anyone to contribute and update Formulae & Casks, and if CI/CD passes and no glaring issues, it will often get automatically merged.  I've done this for various Formulae.  The main hurdle is making sure that any Ruby code passes their rather strict RuboCop configuration, which sometimes can be a bit pedantic with style.  Other than that, it's a pretty open and inclusive community and it's rather easy to contribute and get things merged.

**Reply by LanikSJ** [reply]
Thank you for that insight and I agree it's a great community but like you said they tend to be pedantic about some things. 🙂 

Can't say I blame them it's no easy task to maintain a project of that magnitude and visibility.
